pkg/operator/ceph/object/user/controller.go

```diff
@@ -286,8 +286,31 @@ func (r *ReconcileObjectStoreUser) createorUpdateCephUser(u *cephv1.CephObjectSt
 	// Set access and secret key
 	r.userConfig.Keys[0].AccessKey = user.Keys[0].AccessKey
 	r.userConfig.Keys[0].SecretKey = user.Keys[0].SecretKey
-
 	logger.Info(logCreateOrUpdate)
+
+	if u.Spec.Quotas != nil {
+		if u.Spec.Quotas.MaxObjects != 0 || u.Spec.Quotas.MaxSize != nil {
+			quotaEnable := true
+			var maxSize int64 = -1
+			var maxObjects int64 = -1
+			if u.Spec.Quotas.MaxObjects >= 0 {
+				maxObjects = int64(u.Spec.Quotas.MaxObjects)
+			}
+			if u.Spec.Quotas.MaxSize != nil && !u.Spec.Quotas.MaxSize.IsZero() {
+				maxSize = u.Spec.Quotas.MaxSize.Value()
+			}
+			userQuota := admin.QuotaSpec{
+				UID:        u.Name,
+				Enabled:    &quotaEnable,
+				MaxSize:    &maxSize,
+				MaxObjects: &maxObjects,
+			}
+			err = r.objContext.AdminOpsClient.SetUserQuota(context.TODO(), userQuota)
+			if err != nil {
+				logger.Warningf("failed to set quota for the user %v", u.Name)
```

Comment:
Should we error out here and have the controller reconcile the resource again? I think some admins would consider failure to set up a user's quota as a failure to create the user properly.